x-pack/plugins/watcher/public/lib/use_request.ts

```diff
@@ -55,7 +55,7 @@ export const useRequest = ({
   processData,
 }: UseRequest) => {
   const [error, setError] = useState<null | any>(null);
-  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
```

Comment:
i think we would want the initial state for loading to be `true`, but i might be missing a use case

Reply:
I originally set it to `false` because I was thinking that we may want to add a `sendRequestImmediately` param to this hook, which will allow a developer to use the `createRequest` method that gets returned. For example, we could eventually use this hook to send the "simulate watch" request, which we wouldn't want to represent in a loading state until the user clicks a button, which would call `createRequest`.

Are you running into any issues with defaulting it to `false`?

Reply:
Ah ok. That makes sense. Thanks for explaining.

So do you think this would be the best structure to follow when using this hook?

```
  if (error) {
    // return error component
  }

  if (loading) {
    // return loading component
  }

  if (data) {
    // return main component
  }

  return null;
```

I originally was expecting to return the main component if `loading` and `error` are false, but this causes issues if the initial state of loading is `false`.